cg/services/illumina/backup/backup_service.py

```diff
@@ -278,37 +286,76 @@ def retrieve_archived_file(self, archived_file: Path, run_dir: Path) -> None:
+
+        return validated_responses
+
     @classmethod
-    def get_archived_sequencing_run_path(cls, dsmc_output: list[str]) -> Path | None:
+    def get_latest_archived_sequencing_run_path(cls, dsmc_output: list[str]) -> Path | None:
         """Get the path of the archived sequencing run from a PDC query."""
-        run_line: str = [
-            row
-            for row in dsmc_output
-            if FileExtensions.TAR in row
-            and FileExtensions.GZIP in row
-            and FileExtensions.GPG in row
-        ][0]
-
-        archived_run = Path(run_line.split()[4])
+        validated_sequencing_paths: list[PdcSequencingFile] = cls.parse_dsmc_output_sequencing_path(
+            dsmc_output
+        )
+
+        archived_run: PdcSequencingFile | None = get_latest_pdc_archived_sequencing_run(
+            validated_sequencing_paths
+        )
+
         if archived_run:
             LOG.info(f"Sequencing run found: {archived_run}")
-            return archived_run
+            return archived_run.path
+
+    @staticmethod
+    def parse_dsmc_output_key_path(dsmc_output: list[str]) -> list[PdcEncryptionKey]:
+        """Parses the DSMC command output to extract validated encryption keys."""
+        validated_responses = []
+        for line in dsmc_output:
+            if is_pdc_encryption_key(line):
+                parts: list[str] = line.split()
+
+                fileDateTime: datetime = convert_string_to_datetime_object(
+                    f"{parts[DsmcOutput.DATE_COLUMN_INDEX]} {parts[DsmcOutput.TIME_COLUMN_INDEX]}"
+                )
+
+                query_response: PdcEncryptionKey = PdcEncryptionKey(
+                    dateTime=fileDateTime,
+                    path=Path(parts[DsmcOutput.PATH_COLUMN_INDEX]),
+                )
+
+                validated_responses.append(query_response)
+
+        return validated_responses
 
     @classmethod
-    def get_archived_encryption_key_path(cls, dsmc_output: list[str]) -> Path | None:
+    def get_latest_archived_encryption_key_path(cls, dsmc_output: list[str]) -> Path | None:
         """Get the encryption key for the archived sequencing run from a PDC query."""
-        encryption_key_line: str = [
-            row
-            for row in dsmc_output
-            if FileExtensions.KEY in row
-            and FileExtensions.GPG in row
-            and FileExtensions.GZIP not in row
-        ][0]
-
-        archived_encryption_key = Path(encryption_key_line.split()[4])
+        validated_encryption_keys: list[PdcEncryptionKey] = cls.parse_dsmc_output_key_path(
+            dsmc_output
+        )
+
+        archived_encryption_key: PdcEncryptionKey | None = get_latest_pdc_encryption_key(
+            validated_encryption_keys
+        )
+
         if archived_encryption_key:
             LOG.info(f"Encryption key found: {archived_encryption_key}")
-            return archived_encryption_key
+            return archived_encryption_key.path
```

Comment:
Did we not want to raise an error if the cannot find an encryption key becuase the flow cell is not archived?

Reply:
I agree with you. I talked to Vincent, and he considered that to be scope creep. The error is caught later down the line by the function `retrieve_archived_key`, which also sets `data_availability=SequencingRunDataAvailability.REQUESTED`.

Reply:
Alright, didnt know about the discussion. sounds reasonable.